web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -453,22 +454,38 @@ export function useDiscover<T = any>(): DiscoverContextState<T> {
 }
 
 type BaseMeta = {
-  // resourceName provides a consistent field to refer to for
-  // the resource name since resources can refer to its name
-  // by different field names.
-  // Eg. used in Finish (last step) component.
-  resourceName: string;
-  // agentMatcherLabels are labels that will be used by the agent
-  // to pick up the newly created database (looks for matching labels).
-  // At least one must match.
-  agentMatcherLabels: ResourceLabel[];
+  /**
+   * resourceName provides a consistent field to refer to since
+   * different resources can refer to its name by different field names.
+   * Eg. used in Finish (last step) component.
+   * This field is set when user has finished enrolling a resource.
+   */
+  resourceName?: string;
+  /**
+   * agentMatcherLabels are labels (defined in the enrolled resource)
+   * that are suggested to the user to be used as label matcher for
+   * an agent.
+   *
+   * This field is set when user has finished enrolling a resource.
+   */
+  agentMatcherLabels?: ResourceLabel[];
+
+  /**
+   * awsIntegration is set to the selected AWS integration.
+   * This field is set when a user wants to enroll AWS resources.
+   */
+  awsIntegration?: Integration;
+  /**
+   * awsRegion is set to the selected AWS region.
+   * This field is set when a user wants to enroll AWS resources.
+   */
+  awsRegion?: Regions;
```

Comment:
it would, but it'll be more work to preserve the previous values. like i have to do something like this:
```
update({
  ...
  aws: {
   ...aws,
   region,
  }
})
``` 
integration actually gets set first (they select from existing integration, or came back to the flow from creating a integration) and then next step they have to select a region. or maybe i'm being lazy, what do you think? if you feel strongly i can change it

Reply:
>integration actually gets set first (they select from existing integration, or came back to the flow from creating a integration) and then next step they have to select a region. 

Ok, so these values are not set at the same time. I think we can keep your implementation.